input/otlp/metadata.go

```diff
@@ -38,104 +38,244 @@ var (
 	serviceNameInvalidRegexp = regexp.MustCompile("[^a-zA-Z0-9 _-]")
 )
 
-func translateResourceMetadata(resource pcommon.Resource, out *model.APMEvent) {
+func translateResourceMetadata(resource pcommon.Resource, out *modelpb.APMEvent) {
 	var exporterVersion string
 	resource.Attributes().Range(func(k string, v pcommon.Value) bool {
 		switch k {
 		// service.*
 		case semconv.AttributeServiceName:
+			if out.Service == nil {
+				out.Service = &modelpb.Service{}
+			}
 			out.Service.Name = cleanServiceName(v.Str())
 		case semconv.AttributeServiceVersion:
+			if out.Service == nil {
+				out.Service = &modelpb.Service{}
+			}
```

Comment:
All of this `out.<Field> == nil`  seems a bit verbose. Would it make sense to introduce a generic function that checks if the field is nil, and if it is, it allocates the new type? Something like this would work:

```go
func translateResourceMetadata(resource pcommon.Resource, out *modelpb.APMEvent) {
	[...]
	out.Service = populateNil(out.Service)
	out.Service.Version = truncate(v.Str())
	[...]
}

func populateNil[T any](a *T) *T {
	if a == nil {
		var t T
		return &t
	}
	return a
}
```

Reply:
Thanks a lot for this! :bow: